### PR TITLE
SFR-2137: Integration Tests UofM Ingestion

### PR DESCRIPTION
**Pull request by mitri-slory**
This PR adds integration tests for the UofM ingestion process. This is just the first half of integration tests that will be added for UofM and more will be added after the new oclc process is implemented. The link string can't be properly tested since the has_part attribute which includes the links to the UofM works will be empty locally due to the `addHasPartMapping` method directly calling for the ump-pdf-repository which does not exist locally.

### CODE REVIEW

File: tests/integration/test_uofm_process.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        load_env_file('local', file_string='config/local.yaml')
+        return UofMProcess('complete', None, None, None, None, None, [])
+    
+    @pytest.fixture
+    def test_instance_QA(self):
+        load_env_file('local-qa', file_string='config/local-qa.yaml')
+    
+    def test_runProcess(self, test_instance_QA, test_instance: UofMProcess):
+        # Run process for only 1 record
+        # TODO - create S3 client and upload PDF
```
Comment (kylevillegas93) [inline tests/integration/test_uofm_process.py:22]: we can remove the comments that have been done!

File: ingestJSONFiles/UofM_One_CSV.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            "Digitial copy at press?": "yes",
+            "Status (from Digitization status)": "Publisher digitized",
+            "Digital copy received": "checked",
+            "Project steps": "QA test,On hold",
+            "Next step (from Project steps)": "QA test,On hold",
+            "Date of ingest": "",
+            "Status updated on": "4/8/2024"
+        }
+    ]
+}
```
Comment (kylevillegas93) [inline ingestJSONFiles/UofM_One_CSV.json:29]: let's add a newline here!

File: tests/integration/test_uofm_process.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
+import pytest
+import boto3
+import json
+import os
+from load_env import load_env_file
+from processes import UofMProcess
+from managers import DBManager
+from model import Record
+
+class TestUofMProcess:
```
Comment (kylevillegas93) [inline tests/integration/test_uofm_process.py:10]: let's convert this test to snake case!

File: tests/integration/test_uofm_process.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+        dbManager.generateEngine()
+
+        dbManager.createSession()
+
+        recordExample = dbManager.session.query(Record) \
+            .filter(Record.title == 'Intermediate reading practices : building reading and vocabulary skills ').first()  
+
+        assert recordExample != None
+        assert recordExample.rights == 'UofM|in_copyright||In Copyright|'
```
Comment (kylevillegas93) [inline tests/integration/test_uofm_process.py:63]: Nice this looks great!

File: tests/integration/test_uofm_process.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        # TODO SFR-2150 - kickoff oclc process when it's ready
+        # assert DCDW oclc records exist
+
+        # TODO - kickoff clustering process
+        # assert works, edition, items, links data exist
+
+        # TODO - setup elastic search client
+        # assert search document(s) exist in elastic search
+
+        # TODO - delete data in S3, postgres, and elastic search
```
Comment (kylevillegas93) [inline tests/integration/test_uofm_process.py:74]: we can remove these lines - as part of the OCLC work or after we will create an e2e ingestion -> clustering process

File: tests/integration/test_uofm_process.py (hunk 1 above)
Comment (mitri-slory) [reply tests/integration/test_uofm_process.py:22]: Sure

File: ingestJSONFiles/UofM_One_CSV.json (hunk 2 above)
Comment (mitri-slory) [reply ingestJSONFiles/UofM_One_CSV.json:29]: I can do that

File: tests/integration/test_uofm_process.py (hunk 3 above)
Comment (mitri-slory) [reply tests/integration/test_uofm_process.py:10]: I can do that

File: tests/integration/test_uofm_process.py (hunk 5 above)
Comment (mitri-slory) [reply tests/integration/test_uofm_process.py:74]: That makes sense

File: CHANGELOG.md [hunk 6]
```diff
@@ -22,6 +22,7 @@
 - Added error handling to search API
 - Refactored analytics report code
 - Implemented country-level analytics report
+- Added integration tests for UofM ingestion process
```
Comment (kylevillegas93) [inline CHANGELOG.md:25]: oop! let's add this at the end!

**Review by kylevillegas93 — APPROVED** [review]

**Pull request merged**